File: programming-challenges/mc821/09-20/thinkIwillBuyMeaFootballTeam.py
# TODO this version gets TLE

t = 1
while True:
    n = int(input())
    if n == 0:
        break
    l = []
    for i in range(n):
        l.append([int(x) for x in input().split()])
    cash = 0
    for l_ in l:
        cash += sum(l_)
    opt_cash = 0
    for i in range(len(l)):
        col_sum = 0
        for j in range(len(l)):
            col_sum += l[j][i]
        diff = col_sum - sum(l[i])
        opt_cash += max(diff, 0)
    print("{}. {} {}".format(t, cash, opt_cash))
    t += 1

File: programming-challenges/mc821/09-20/thinkIwillBuyMeaFootballTeam.c
#include <stdio.h>

int col_sum[1000] = {0};
int row_sum[1000] = {0};

int main(){
    int n, i, j, x;
    int t = 1;
    while(1){
        scanf("%d", &n);
        if(n == 0) break;
        for(i = 0; i < n; i++){
            col_sum[i] = row_sum[i] = 0;
        }
        int cash = 0;
        for(i = 0; i < n; i++){
            for(j = 0; j < n; j++){
                scanf("%d", &x);
                cash += x;
                row_sum[i] += x;
                col_sum[j] += x;
            }
        }
        int diff;
        int opt_cash = 0;
        for(i = 0; i < n; i++){
            diff = col_sum[i] - row_sum[i];
            if(diff > 0) opt_cash += diff;
        }
        printf("%d. %d %d\n", t, cash, opt_cash);
        t++;
    }
    return 0;
}
